offer-manager-domain/src/main/groovy/life/qbic/business/offers/Offer.groovy

```diff
@@ -447,11 +444,34 @@ class Offer {
     }
 
     private double calculateNetPrice() {
-        double netSum = 0.0
-        for (item in items) {
-            netSum += item.quantity * item.product.unitPrice
+        double netPrice
+        switch (selectedCustomerAffiliation.category) {
+            case AffiliationCategory.INTERNAL:
+                netPrice = calculateInternalNetPrice()
+                break
+            default:
+                netPrice = calculateExternalNetPrice()
+                break
         }
-        return netSum
+        // for backwards compatibility, we need to add the product item cost of product items that have no internal or external costs
+        netPrice += items.collect {it.product.unitPrice * it.quantity}.sum() as double
```

Comment:
I think this is confusing. I would rather prefer to have some logic which tests if internal and external price are set correctly (are not null?) and then trigger the correct calculation. Now it seems like we calculate all cases every time which makes no sense to me.

Reply:
we would have to check for every single item. Doing that would prevent us from splitting the method into the external and internal computation.

Reply:
Why? I would assume that its actually an offer state if the prices are internal or external, right? Can the states between the product items differ?

Reply:
yes. You can select old products and new ones I think